src/components/Regridder.vue

```diff
@@ -0,0 +1,232 @@
+<template>
+    <div> 
+      <!-- Download Dialog -->
+      <v-dialog v-model="downloadDialog" max-width="500">
+        <v-card>
+          <v-card-title>Download netCDF Data</v-card-title>
+          <v-card-text>
+            <v-form>
+              <v-text-field v-model="dx" label="dx" required></v-text-field>
+              <v-text-field v-model="dy" label="dy" required></v-text-field>
+            </v-form>
+          </v-card-text>
+          <v-card-actions>
+            <v-btn color="primary" @click="downloadClicked">Download</v-btn>
+            <v-btn color="error" @click="closeDownloadDialog">Close</v-btn>
+          </v-card-actions>
+        </v-card>
+      </v-dialog>
+  
+      <!-- Error Dialog -->
+      <v-dialog v-model="errorDialog" max-width="500">
+        <v-card>
+          <v-card-title>Error</v-card-title>
+          <v-card-text>{{ errorMessage }}</v-card-text>
+          <v-card-actions>
+            <v-btn color="error" @click="closeErrorDialog">Close</v-btn>
+          </v-card-actions>
+        </v-card>
+      </v-dialog>
+    </div>
+</template>
+  
+
+<script lang="ts">
+import { Component, Inject, Mixins, Prop} from 'vue-property-decorator'
+import { Map } from 'mapbox-gl'
+import MapboxDraw from '@mapbox/mapbox-gl-draw'
+import DrawRectangle from 'mapbox-gl-draw-rectangle-mode'
+import {DownloadControl} from "../lib/DownloadControl"
+import { PiWebserviceProvider} from "@deltares/fews-pi-requests";
+import PiRequestsMixin from '@/mixins/PiRequestsMixin';
+
+@Component
+export default class Regridder extends Mixins(PiRequestsMixin) {
+    @Inject() getMap!: () => Map
+
+    @Prop() firstValueTime!: string
+    @Prop() lastValueTime!: string
+
+    mapObject!: Map
+    isInitialized = false
+    draw!: MapboxDraw
+    downloadControl: DownloadControl | undefined
```

Comment:
In my experience, reactivity on `undefined` variables will often confuse Vue and lead to errors. Using `null` solves these problems, as Vue then realises that the variable actually does exist but is just intentionally not set.